### PR TITLE
docs: shared `pytket-docs-theming` submodule

### PR DESCRIPTION
**Pull request by CalMacCQ**
# Description

Updating the manual and examples build to use `pytket-docs-theming` as a submodule. I've already done the same for all of the extensions.

This ensures that the `_static` folder, quantinuum-sphinx version and sphinx `conf.py` are in a single location and can be easily updated.

# Related issues

Please mention any github issues addressed by this PR.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -22,6 +22,8 @@ openfermion = "^1.6.1"
 ipyparallel = "^8.8.0"
 furo = "^2024.8.6"
 myst-nb = "^1.1.2"
+sphinx-autodoc-typehints = "^2.4.4"
```
Comment (CalMacCQ) [inline pyproject.toml:25]: These dependencies aren't really used here but I've added them as they are also used by the extensions docs. See the `extensions` list variable in `conf.py` maybe I could get around this by altering the extensions list depending on what the parent repository is but that didn't seem very nice.

File: build-docs.sh [hunk 2]
```diff
@@ -3,5 +3,15 @@
 cd docs/
 
 rm -rf build/
-sphinx-build -b html . build -W
+
+cp -R pytket-docs-theming/_static .
+cp -R pytket-docs-theming/quantinuum-sphinx .
+cp pytket-docs-theming/conf.py .
+
+sphinx-build -b html . build -D html_title="pytket user guide"
```
Comment (cqc-alec) [inline build-docs.sh:11]: Can we keep the `-W` (which I assume treats warnings as errors)?

Comment (CalMacCQ) [reply build-docs.sh]: added this back in after resolving some warnings in [6a6ab33](https://github.com/CQCL/pytket-docs/pull/389/commits/6a6ab33243d891f24ba6afea2e1936b23fbbc4f4) and [1691730](https://github.com/CQCL/pytket-docs/pull/389/commits/16917309e0fd05e29216a1c20c71a2561ef198f3).

File: docs/manual/manual_circuit.md [hunk 3]
```diff
@@ -622,7 +622,7 @@ print("Total gate count =", circ.n_gates)
 print("Circuit depth =", circ.depth())
 ```
 
-As characteristics of a {py:class}`~pytket.circuit.Circuit` go, these are pretty basic. In terms of approximating the noise level, they fail heavily from weighting all gates evenly when, in fact, some will be much harder to implement than others. For example, in the NISQ era, we find that most technologies provide good single-qubit gate times and fidelities, with two-qubit gates being much slower and noisier [^cite_arut2019]. On the other hand, looking forward to the fault-tolerant regime we will expect Clifford gates to be very cheap but the magic $T$ gates to require expensive distillation procedures [^cite_brav2005] [^cite_brav2012].
```
Comment (CalMacCQ) [inline docs/manual/manual_circuit.md:625]: I was getting a warning that this citation doesn't appear in the refernces. Just removed it for now as I'm not sure which paper it was.

**Pull request merged**